src/app/features/tasks/task.service.ts

```diff
@@ -753,6 +753,15 @@ export class TaskService {
     ) as Task[];
   }
 
+  async getArchivedTasks(): Promise<Task[]> {
+    const archiveTaskState: TaskArchive = await this._persistenceService.taskArchive.loadState(
+      true,
+    );
```

Comment:
I'm passing _true_ to loadState() now. Otherwise, for every time the user switches task source while searching, the migration process gets triggered...

Reply:
Good choice! Tbh. I forgot about this mechanism for loadState completely.  :)